tsdb/index.go

```diff
@@ -1104,20 +1105,40 @@ func (is IndexSet) Database() string {
 	return is.Indexes[0].Database()
 }
 
-// FieldSet returns the fieldset of the first index.
-func (is IndexSet) FieldSet() *MeasurementFieldSet {
+// HasField determines if any of the field sets on the set of indexes in the
+// IndexSet have the provided field for the provided measurement.
+func (is IndexSet) HasField(measurement []byte, field string) bool {
 	if len(is.Indexes) == 0 {
-		return nil
+		return false
+	}
+
+	if len(is.fieldSets) == 0 {
+		// field sets may not have been initialised yet.
+		is.fieldSets = make([]*MeasurementFieldSet, 0, len(is.Indexes))
```

Comment:
This looks racy.  Couldn't two concurrent queries end up assigning to `is.fieldSets` at the same time?

Reply:
An `IndexSet` is not  (shouldn't be anyway) shared across goroutines. It should be one per query.

Reply:
Also, this is on a value receiver, so assignments to the field won't leak anyway... :-)

Reply:
It also means nothing is being cached.